src/AutoRest.CSharp/Common/Output/Models/Types/EnumType.cs

```diff
@@ -55,15 +55,21 @@ public EnumType(InputEnumType input, string defaultNamespace, string defaultAcce
             }
 
             ValueType = typeFactory.CreateType(input.EnumValueType);
+            IsStringValueType = ValueType.Equals(typeof(string));
+            IsIntValueType = ValueType.Equals(typeof(int)) || ValueType.Equals(typeof(long));
+
             Description = input.Description;
             IsExtensible = isExtensible;
         }
 
         public CSharpType ValueType { get; }
         public bool IsExtensible { get; }
-        public bool IsStringValueType => ValueType.Equals(typeof(string));
-        public bool IsIntValueType => ValueType.Equals(typeof(Int32)) || IsLongValueType;
-        public bool IsLongValueType => ValueType.Equals(typeof(Int64));
+        public bool IsStringValueType { get; }
+        public bool IsIntValueType { get; }
+
+        private string? _serializationMethodName;
+        public string SerializationMethodName => _serializationMethodName ??= IsStringValueType && IsExtensible ? "ToString" : $"ToSerial{ValueType.Name.FirstCharToUpperCase()}";
```

Comment:
Both `IsStringValueType` and `IsExtensible` values are set in constructor, hence `SerializationMethodName` can be set in constructor as well.

Reply:
make sense. Let me change this